rholang/src/test/scala/coop/rchain/rholang/interpreter/accounting/CostAccountingSpec.scala

```diff
@@ -89,6 +105,31 @@ class CostAccountingSpec extends FlatSpec with Matchers with PropertyChecks with
     }
   }
 
+  it should "be repeatable" in
+    forAll(contracts) { (contract: String, _) =>
+      checkRepeatableCost {
+        val result = evaluateWithCostLog(Integer.MAX_VALUE, contract)
+        assert(result._1.errors.isEmpty)
+        result
+      }
+    }
+
+  def checkRepeatableCost(block: => (EvaluateResult, Chain[Cost])): Unit = {
+    val repetitions = 20
+    val first       = block
+    // execute in parallel to trigger different interleaves
+    val subsequents = (1 to repetitions).par.map(_ => block).toList
```

Comment:
What's the point of parallelizing this? This creates 20 different runtimes and never writes to disk. Removing the par also prevents the weird errors you're getting in the tests from materializing.

Reply:
To make it run faster. Also, it's easier to observe cost non-determinism that way (the 'keccak' test didn't fail for me even with 100 repetitions without the `.par`)

Reply:
I get the performance argument, but not the non-determinism argument. Each run uses a different hot store, so why is it easier to observe non-determinism?

Reply:
my observation was the test doesn't catch cost nondeterminism otherwise. my intution is having more tasks than cores is needed for non-as-if-sequential interleaves to happen.